crypto/objects/obj_compat.h

```diff
@@ -1,3 +1,11 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
```

Comment:
This is the cause of the "make update" weirdness. This file is not actually a standalone headerfile. obj_mac.h gets auto generated by objects.pl, and then this file gets concatenated onto the end of it.

So, I'm thinking that we should revert the change in this file?

Reply:
Then there'd be no copyright in that file...

Reply:
so what?

Reply:
It means that file is not subject to the Apache License, with all subsequent ramifications, thereof.
The project wanted to be under the Apache License 2.0. One could argue that if there's no license on a file, that someone could subsequently put a different license on a file, including GPL, and all that entails.

Reply:
but **this** is a super-ugly hack:
`Configurations/unix-Makefile.tmpl:	( cd $(SRCDIR);	cat crypto/objects/obj_compat.h >> include/openssl/obj_mac.h )`
why don't we just do this `echo #include "obj_compat.h" >> include/openssl/obj_mac.h )`, and move that file
to include/openssl/ ?

Reply:
for a different PR of course.

Reply:
I agree. But the extra comment doesn't hurt. However, adding/moving a new header file changes things a bit more. I know a few build systems that would choke on a new header file being added (not that they couldn't be fixed).

Reply:
... or just do this `	( cd $(SRCDIR); grep "^#" crypto/objects/obj_compat.h >> include/openssl/obj_mac.h )`

Reply:
Or `( cd $(SRCDIR); tail +8 crypto/objects/obj_compat.h >> include/openssl/obj_mac.h )`

Reply:
note that `tail +8` does not work on all versions:
`'`
$ tail +8                             
tail: cannot open '+8' for reading: No such file or directory
$ tail --version 
tail (GNU coreutils) 8.21
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Paul Rubin, David MacKenzie, Ian Lance Taylor,
and Jim Meyering.
```

Reply:
I think you might need to instead do `tail -n +8` on some versions (works with my version too)

The same can be achieved using sed/awk etc, e.g. see https://unix.stackexchange.com/questions/37790/how-do-i-delete-the-first-n-lines-of-an-ascii-file-using-shell-commands

Reply:
It does not appear that `awk` is used outside of `util/perl/OpenSSL/config.pm` but `sed` is used inside `Configurations/unix-Makefile.tmpl`, so that seems a bit safer. `tail` is also not used in the build system (AFAICT), and I don't want to introduce any new system dependencies.

Reply:
Fixed with a `sed -e '1,8d'`, which removes the first 8 lines of the file, which is *just* the copyright. So it should not impact further modification of the file.

Reply:
It'll be interesting if we change this boilerplate some day, enough to add, say, one extra line :wink: